qp_shogun/sortmerna/sortmerna.py

```diff
@@ -79,29 +78,73 @@ def generate_sortmerna_commands(forward_seqs, reverse_seqs, map_file,
 
     cmds = []
     param_string = _format_params(parameters, SORTMERNA_PARAMS)
+    threads = parameters['Number of threads']
 
-    # Sortmerna 2.1 does not support processing of
-    # compressed files but they said the newest release might
-    # but that version first has to be tested before use and currently
-    # does not have MAC OS supported release
+    # Sortmerna 2.1 does not support direct processing of
+    # compressed files currently
 
     for run_prefix, sample, f_fp, r_fp in samples:
-        cmds.append('sortmerna --ref %s --reads %s '
```

Comment:
For the command executions using `&&` instead of semicolons is preferable. This will help us identify errors on the commands that fail. Otherwise if `unpigz` failed we would probably see an error from `sortmerna` saying that the input file was not found or something along those lines.

Second, since the command is pretty much the same for forward and reverse files, I suggest setting that as a template and doing the formatting via a dictionary. For example like this:

```python
template = ("unpigz -p {threads -c {ip} > {ip_unpigz} && "
            "sortmerna --ref {ref_db} --reads {ip_unpigz} ...")
arguments = {'threads': threads, 'ref_db': RNA_REF_DB}

for run_prefix, sample, f_fp, r_fp in samples:
    prefix_path = join(out_dir, run_prefix)

    for index, fp in enumerate([f_fp, r_fp]):
        # if the reverse filepath is not present ignore it
        if fp is None:
            continue

        arguments['ip'] = fp
        arguments['ip_unpigz'] = fp.replace('.fastq.gz', '.fastq')
        arguments['smr_r_op'] = prefix_path + '.ribosomal.R%d' % (index + 1)
        arguments['smr_nr_op'] = prefix_path + '.nonribosomal.R%d' % (index + 1)
        arguments['smr_r_op_gz'] = arguments['smr_r_op'] + '.gz'
        arguments['smr_nr_op_gz'] = arguments['smr_nr_op'] + '.gz'

        cmds.append(template.format(**arguments))
```

Reply:
Ah, I guess I saw this after I pushed the modified code to update memory requirements.

Reply:
Ping on this comment.

Reply:
Done!